src/go/rpk/pkg/cli/cmd/debug/bundle/bundle_k8s_linux.go

```diff
@@ -307,3 +350,63 @@ func requestAndSave[T1 any](ctx context.Context, ps *stepParams, filename string
 	}
 	return nil
 }
+
+func parseJournalTime(t string) (time.Time, error) {
+	/*
+		From `man journalctl`:
+			Date specifications should be of the format "2012-10-30 18:17:16". If the time part is omitted, "00:00:00" is assumed.
+			If only the seconds component is omitted, ":00" is assumed. If the date component is omitted, the current day is assumed.
+			Alternatively the strings "yesterday", "today", "tomorrow" are understood, which refer to 00:00:00 of the day before
+			the current day, the current day, or the day after the current day, respectively. "now" refers to the current time.
+			Finally, relative times may be specified, prefixed with "-" or "+", referring to times before or after the current
+			time, respectively.
+	*/
+
+	// First we ensure that we don't have any leading/trailing whitespace.
+	t = strings.TrimSpace(t)
+
+	// Will match YYYY-MM-DD, where Y,M and D are digits.
+	ymd := regexp.MustCompile(`^\d{4}-\d{2}-\d{2}`)
+
+	// Will match YYYY-MM-DD HH:MM:SS or YYYY-MM-DD HH:MM
+	//   - index 0: the full match.
+	//   - index 1: the seconds, if present.
+	ymdOptionalSeconds := regexp.MustCompile(`^\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}(:\d{2})?`)
+
+	switch {
+	// YYYY-MM-DD
+	case len(t) == 10 && ymd.MatchString(t):
+		return time.ParseInLocation("2006-01-02", t, time.Local)
+
+	// YYYY-MM-DD HH:MM:SS or YYYY-MM-DD HH:MM
+	case ymdOptionalSeconds.MatchString(t):
+		m := ymdOptionalSeconds.FindStringSubmatch(t)
+		if len(m) == 0 {
+			return time.Time{}, errors.New("is this even possible? please file a bug")
```

Comment:
Please don't merge until we solve this part/ return a better error/ remove the nil check

AFAICT there is no way that at this point the `len(m) == 0` since the case already matches the string for the same regex, I added this out of paranoia but wanted to confirm if it's ok to remove this nil check: @twmb

Reply:
I'd move the compiling and matching above the switch:
```go
ymd := regexp.MustCompile(`^\d{4}-\d{2}-\d{2}`).FindStringSubmatch(t)
ymdOptSec := regexp.MustCompile(`^\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}(:\d{2})?`).FindStringSubmatch(t)
switch {
case len(ymd) > 0:
case len(ymdOptSec) > 0:
...
```